Auth/OpenID/Parse.php

```diff
@@ -218,12 +218,12 @@ function removeQuotes($str)
     
     function match($regexp, $text, &$match)
     {
-        if (!is_callable('mb_ereg_search_init')) {
-            if (!preg_match($regexp, $text, $match)) {
-                return false;
-            }
-            $match = $match[0];
-            return true;
+        if (preg_match($regexp, $text, $match)) {
+           return true;
+        }
+
+        if ( !is_callable('mb_ereg_search_init')) {
```

Comment:
Also, since you're basically short circuiting the whole method, you can probably just kill lines 229 - 235. If preg_match works as expected.